src/implementation/Client/HTTPClient/invoker.ts

```diff
@@ -23,20 +26,24 @@ export default class HTTPClientInvoker implements IClientInvoker {
     this.client = client;
   }
 
-  async invoke(appId: string, methodName: string, method: HttpMethod = HttpMethod.GET, data?: object): Promise<object> {
+  async invoke(appId: string, methodName: string, method: HttpMethod = HttpMethod.GET, data?: object, options: InvokerOptions = {}): Promise<object> {
+
+    const headers = options.headers ?? {}
+ 
     const fetchOptions = {
       method,
+
+      headers
     };
 
     if (method !== HttpMethod.GET) {
-      // @ts-ignore
-      fetchOptions.headers = {
-        "Content-Type": "application/json",
-      };
+       //@ts-ignore
+      fetchOptions.headers['Content-Type'] = 'application/json';
+
```

Comment:
Since we are allowing users to pass `headers`, we can skip setting `Content-Type` if it's already set. WDYT @XavierGeerinck?